services/hh/processor.go

```diff
@@ -23,7 +23,20 @@ const (
 	bytesWrite  = "bytes_write"
 )
 
+// MetaProvider specifies the interface of a module that the Processor may query
+// to determine active node IDs.
+type MetaProvider interface {
+	// Nodes returns the active cluster node IDs
+	Nodes() []uint64
+}
+
+// Processor manages the loading and processing of hinted data. If MetaProvider is non-nil
+// it is queried for active nodes during processing. In this case only if a node ID is active
+// will it be processed. Data for inactive nodes remain on disk, and the purge process must
```

Comment:
Originally, https://github.com/influxdb/influxdb/blob/master/services/hh/service.go#L168-L172 was planned to run as a groutine and purge them more aggressively.  It was going to delete queues for nodes that did not exists as well as shards that are gone.